src/tests/unit/tests/DetailsView/components/start-over-dropdown.test.tsx

```diff
@@ -65,67 +76,72 @@ describe('StartOverDropdownTest', () => {
-        expect(rendered.getElement()).toMatchSnapshot();
+
+        const renderResult = render(<StartOverDropdown {...props} />);
+        getMockComponentClassPropsForCall(InsightsCommandButton).onClick({
+            currentTarget: 'test target',
+        });
+        expect(renderResult.asFragment()).toMatchSnapshot();
     });
 
-    it('should open the start test over dialog', () => {
+    it('should open the start test over dialog', async () => {
         openDialogMock.setup(sds => sds('test')).verifiable(Times.once());
 
-        const rendered = shallow<StartOverDropdown>(<StartOverDropdown {...defaultProps} />);
-        rendered.find(InsightsCommandButton).simulate('click', event);
-        rendered
-            .find(ContextualMenu)
-            .prop('items')
-            .find(elem => elem.key === 'test')
+        render(<StartOverDropdown {...defaultProps} />);
+        getMockComponentClassPropsForCall(InsightsCommandButton).onClick({
+            currentTarget: 'test target',
+        });
+        getMockComponentClassPropsForCall(ContextualMenu)
+            .items.find(elem => elem.key === 'test')
             .onClick();
 
         openDialogMock.verifyAll();
     });
 
-    it('should open the start assessment over dialog', () => {
+    it('should open the start assessment over dialog', async () => {
         openDialogMock.setup(sds => sds('assessment')).verifiable(Times.once());
 
-        const rendered = shallow<StartOverDropdown>(<StartOverDropdown {...defaultProps} />);
-        rendered.find(InsightsCommandButton).simulate('click', event);
-        rendered
-            .find(ContextualMenu)
-            .prop('items')
-            .find(elem => elem.key === 'assessment')
+        render(<StartOverDropdown {...defaultProps} />);
+        getMockComponentClassPropsForCall(InsightsCommandButton).onClick({
+            currentTarget: 'test target',
+        });
+        getMockComponentClassPropsForCall(ContextualMenu)
+            .items.find(elem => elem.key === 'assessment')
             .onClick();
 
         openDialogMock.verifyAll();
     });
 
-    it('should dismiss the contextMenu', () => {
-        const rendered = shallow<StartOverDropdown>(<StartOverDropdown {...defaultProps} />);
-        rendered.find(InsightsCommandButton).simulate('click', event);
-        rendered.find(ContextualMenu).prop('onDismiss')();
-
-        expect(rendered.state().isContextMenuVisible).toBe(false);
-        expect(rendered.state().target).toBeNull();
+    it('should dismiss the contextMenu', async () => {
+        const renderResult = render(<StartOverDropdown {...defaultProps} />);
+        getMockComponentClassPropsForCall(InsightsCommandButton).onClick({
+            currentTarget: 'test target',
+        });
+        getMockComponentClassPropsForCall(ContextualMenu).onDismiss();
+        const mockContextualmenu = renderResult.container.querySelector('mock-contextual-menu');
```

Comment:
Adjusting mock tag name to match the snapshot
```suggestion
        const mockContextualmenu = renderResult.container.querySelector('mock-contextualmenu');
```